tools/c7n_azure/c7n_azure/provider.py

```diff
@@ -37,8 +37,9 @@ def initialize(self, options):
         self.cloud_endpoints = self._get_cloud_endpoints(options)
         options['region'] = self.cloud_endpoints.name
 
+        # c7n-org needs to initialize a session to set the correct cloud endpoints
+        session = local_session(self.get_session_factory(options))
```

Comment:
Any reason it needs to be done here vs initialize in `c7n-org` directly?

Reply:
`get_session_factory` needed to be called for c7n-org so the `cloud_endpoints` can be set correctly for the azure session. This line of code was within the `if options['account_id'] is None:` check which is populated already for c7n-org.

Is there a way to make sure the session is created properly within c7n-org itself instead of having to do this?

Reply:
session will be created automatically first time code tries to get session.. So It will be instantiated at some point.
Also, session object is tight to the subscription, so `c7n-org` will create multiple sessions regardless

Reply:
Interestingly, when I was debugging I set a breakpoint at this location and c7n-org never reaches this initialization.  During c7n-org initialization, I did see session creation for AWS, but nothing related to azure. I think I may be misunderstanding how c7n-org creates a session.